### PR TITLE
[BF kNN] Minkowski distance optimizations

### PR DESCRIPTION
**Pull request by KalyanovD**
# Description
\>1.5x against scikit-learn on real datasets.

Includes #2072 optimizations

### CODE REVIEW

**Comment by KalyanovD** [comment]
http://intel-ci.intel.com/ec93fe0e-fc88-f1bc-9646-8cdcd4b723a1

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 1]
```diff
@@ -531,6 +531,110 @@ class ChebyshevDistances : public PairwiseDistances<FPType, cpu>
...
+    float * tmp            = new float[vecSize];
+    const __m512 * ptr512x = (__m512 *)x;
+    const __m512 * ptr512y = (__m512 *)y;
+
+    if (_p == 1.0)
+    {
+        size_t i = 0;
+        for (; i < nBlocks; ++i)
+        {
+            d += _mm512_reduce_add_ps(_mm512_abs_ps(_mm512_sub_ps(ptr512x[i], ptr512y[i])));
```
Comment (makart19) [inline cpp/daal/src/algorithms/service_kernel_math.h:553]: As next step please try using simple _mm512_add_ps in loop into __mm512 variable , and make only one horizontal reduce after the loop.

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 2]
```diff
@@ -531,6 +531,110 @@ class ChebyshevDistances : public PairwiseDistances<FPType, cpu>
...
+        }
+
+        for (i *= vecSize; i < n; ++i)
```
Comment (makart19) [inline cpp/daal/src/algorithms/service_kernel_math.h:556]: try using mask intrinsics instead of this loop.

**Review by makart19 — APPROVED** [review]

**Pull request merged**